conn.go

```diff
@@ -30,12 +31,52 @@ type closeError struct {
 
 func (e *closeError) Error() string {
 	var cleanStmt string
+	var reason string // cant override the event's reason
 	if e.WasClean {
 		cleanStmt = "clean"
 	} else {
 		cleanStmt = "unclean"
+
+		if e.Reason != "" {
+			console.Warn("original Reason:", e.Reason)
+			reason = e.Reason
+		}
+
+		// taken from http://stackoverflow.com/a/28396165
+		// See http://tools.ietf.org/html/rfc6455#section-7.4.1
+		switch e.Code {
+		case 1000:
+			reason = "Normal closure, meaning that the purpose for which the connection was established has been fulfilled."
+		case 1001:
+			reason = "An endpoint is \"going away\", such as a server going down or a browser having navigated away from a page."
+		case 1002:
+			reason = "An endpoint is terminating the connection due to a protocol error"
+		case 1003:
+			reason = "An endpoint is terminating the connection because it has received a type of data it cannot accept (e.g., an endpoint that understands only text data MAY send this if it receives a binary message)."
+		case 1004:
+			reason = "Reserved. The specific meaning might be defined in the future."
+		case 1005:
+			reason = "No status code was actually present."
+		case 1006:
+			reason = "The connection was closed abnormally, e.g., without sending or receiving a Close control frame"
+		case 1007:
+			reason = "An endpoint is terminating the connection because it has received data within a message that was not consistent with the type of the message (e.g., non-UTF-8 [http://tools.ietf.org/html/rfc3629] data within a text message)."
+		case 1008:
+			reason = "An endpoint is terminating the connection because it has received a message that \"violates its policy\". This reason is given either if there is no other sutible reason, or if there is a need to hide specific details about the policy."
+		case 1009:
+			reason = "An endpoint is terminating the connection because it has received a message that is too big for it to process."
+		case 1010: // Note that this status code is not used by the server, because it can fail the WebSocket handshake instead
+			reason = "An endpoint (client) is terminating the connection because it has expected the server to negotiate one or more extension, but the server didn't return them in the response message of the WebSocket handshake. <br /> Specifically, the extensions that are needed are: " + reason
+		case 1011:
+			reason = "A server is terminating the connection because it encountered an unexpected condition that prevented it from fulfilling the request."
+		case 1015:
+			reason = "The connection was closed due to a failure to perform a TLS handshake (e.g., the server certificate can't be verified)."
+		default:
+			reason = "Unknown reason"
+		}
```

Comment:
Instead of having this switch here, consider having an unexported map of codes to reasons outside, and this can reference that map. Maybe it would be better?